File: AdvantageFramework/Email/Classes/Attachment.vb
Namespace Email.Classes

    <Serializable()>
    Public Class Attachment

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            AttachmentName
            File
            FileBytes
        End Enum

#End Region

#Region " Variables "

        Private _AttachmentName As String = ""
        Private _File As String = ""
        Private _FileBytes() As Byte = Nothing

#End Region

#Region " Properties "

        <System.Runtime.Serialization.DataMemberAttribute()>
        Public Property AttachmentName() As String
            Get
                AttachmentName = _AttachmentName
            End Get
            Set(ByVal value As String)
                _AttachmentName = value
            End Set
        End Property
        <System.Runtime.Serialization.DataMemberAttribute()>
        Public Property File() As String
            Get
                File = _File
            End Get
            Set(ByVal value As String)
                _File = value
            End Set
        End Property
        <System.Runtime.Serialization.DataMemberAttribute()>
        Public Property FileBytes() As Byte()
            Get
                FileBytes = _FileBytes
            End Get
            Set(ByVal value As Byte())
                _FileBytes = value
            End Set
        End Property

#End Region

#Region " Methods "

        Public Sub New(ByVal File As String)

            _File = File

        End Sub
        Public Sub New(ByVal AttachmentName As String, ByVal File As String)

            _AttachmentName = AttachmentName
            _File = File

        End Sub
        Public Sub New(ByVal AttachmentName As String, ByVal FileBytes() As Byte)

            _AttachmentName = AttachmentName
            _FileBytes = FileBytes

        End Sub

#End Region

    End Class

End Namespace

File: AdvantageFramework.Core/Email/Classes/Attachment.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdvantageFramework.Core.Email.Classes
{
    [Serializable()]
    public partial class Attachment
    {

        /* TODO ERROR: Skipped RegionDirectiveTrivia */


        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        public enum Properties
        {
            AttachmentName,
            File,
            FileBytes
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private string _AttachmentName = "";
        private string _File = "";
        private byte[] _FileBytes = null;

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        [System.Runtime.Serialization.DataMemberAttribute()]
        public string AttachmentName
        {
            get
            {
                string AttachmentNameRet = default;
                AttachmentNameRet = _AttachmentName;
                return AttachmentNameRet;
            }

            set
            {
                _AttachmentName = value;
            }
        }

        [System.Runtime.Serialization.DataMemberAttribute()]
        public string File
        {
            get
            {
                string FileRet = default;
                FileRet = _File;
                return FileRet;
            }

            set
            {
                _File = value;
            }
        }

        [System.Runtime.Serialization.DataMemberAttribute()]
        public byte[] FileBytes
        {
            get
            {
                byte[] FileBytesRet = default;
                FileBytesRet = _FileBytes;
                return FileBytesRet;
            }

            set
            {
                _FileBytes = value;
            }
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        public Attachment(string File)
        {
            _File = File;
        }

        public Attachment(string AttachmentName, string File)
        {
            _AttachmentName = AttachmentName;
            _File = File;
        }

        public Attachment(string AttachmentName, byte[] FileBytes)
        {
            _AttachmentName = AttachmentName;
            _FileBytes = FileBytes;
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }
}
